index.mjs

```diff
@@ -44,19 +45,26 @@ export function $(pieces, ...args) {
     }
     cmd += s + pieces[++i]
   }
-  if (verbose) {
-    printCmd(cmd)
-  }
-  let options = {
-    cwd: $.cwd,
-    shell: typeof $.shell === 'string' ? $.shell : true,
-    windowsHide: true,
-  }
-  let child = spawn($.prefix + cmd, options)
-  let promise = new ProcessPromise((resolve, reject) => {
+
+  let resolve, reject
+  let promise = new ProcessPromise((...args) => [resolve, reject] = args)
+
+  promise._run = () => {
+    if (promise.child) return
+    if (promise._prerun) promise._prerun()
+    if (verbose) {
+      printCmd(cmd)
+    }
+
+    let child = spawn(prefix + cmd, {
+      cwd,
+      shell: typeof shell === 'string' ? shell : true,
+      stdio: [promise._inheritStdin ? 'inherit' : 'pipe', 'pipe', 'pipe'],
```

Comment:
Why use 'inherit' instead of 'pipe' for this parameter.

in cases like this, errors are reported：
```
#!/usr/bin/env zx
const sleep = () => new Promise((resolve) => setTimeout(resolve, 1000));

let p = $`read var; echo "$var";`

await sleep();

p.stdin.write('Hello, stdin!\n')  //  p.stdin is null
```

Reply:
So, now zx comes with piped stdin by default.